pkg/cmd/marathon/app/app_add.go

```diff
@@ -1,18 +1,52 @@
 package app
 
 import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/dcos/dcos-core-cli/pkg/marathon"
+
 	"github.com/dcos/dcos-cli/api"
-	"github.com/dcos/dcos-core-cli/pkg/cmd/marathon/python"
 	"github.com/spf13/cobra"
 )
 
 func newCmdMarathonAppAdd(ctx api.Context) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "add",
 		Short: "Add an application.",
-		RunE: func(cmd *cobra.Command, args []string) error {
-			return python.InvokePythonCLI(ctx)
+		Args:  cobra.MaximumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			client, err := marathon.NewClient(ctx)
+			if err != nil {
+				fmt.Fprintf(ctx.ErrOut(), "%s", err)
+				return
+			}
+			if len(args) > 0 {
+				err = marathonAppAdd(ctx, *client, args[0])
+			} else {
+				err = marathonAppAdd(ctx, *client, "")
+			}
+			if err != nil {
+				os.Exit(1)
+			}
 		},
 	}
 	return cmd
 }
+
+func marathonAppAdd(ctx api.Context, client marathon.Client, appFile string) error {
+	app, err := client.AddApp(ctx, appFile)
+	if err != nil {
+		if err == marathon.ErrCannotReadAppDefinition {
+			fmt.Fprintf(ctx.ErrOut(), "Can't read from resource: %s.\nPlease check that it exists.\n", appFile)
+		} else if syntaxError, ok := err.(*json.SyntaxError); ok {
+			fmt.Fprintf(ctx.ErrOut(), "Error loading JSON: %s\n", syntaxError.Error())
+		} else {
+			fmt.Fprintf(ctx.ErrOut(), "%s\n", err.Error())
```

Comment:
Look at my comment above. All this `fmt.Printf` should be changed to `return fmt.Errorf`
```suggestion
			return fmt.Errorf(ctx.ErrOut(), "%s\n", err.Error())
```